src/lib/types/restrictions.ts

```diff
@@ -77,7 +85,8 @@ const fixedOutputMetadataValueType = z.object({
   metadataValueType: metadataValueType,
 })
 
-export const userRestrictionValidation = z.object({
+export const userRestrictionValidation = z.array(z.object({
+  op: binaryOperator.optional(),
```

Comment:
I'm not sure I like this being at the same level as the restriction types. Can we look to mirror the transaction format better?

Reply:
@mattdean-digicatapult  i've chosen an array because `node` expect it to be an array in the format of 
```js
[{ Restriction: {}, { op: 'AND' }, ...]
```
. Happy to reformat to something less confusing. Any chance you have it in your mind?

Reply:
So the outer array is good and I like that. There are two improvements I'd suggest. Firstly each restriction type has an additional level of array, for example:
```
  SenderOwnsAllInputs: z.array(senderOwnsAllInputs).optional(),
```
I don't think these should be arrays.

Second I think we should look at separating `Op` and `Restriction` to separate levels so it does look more like the transaction format:
```
[{ Restriction: {}, { op: 'AND' }, ...]
```

Reply:
This is again has been updated. I'm going to mark as resolved so it's easier for me to track, feel free to unresolve if needed